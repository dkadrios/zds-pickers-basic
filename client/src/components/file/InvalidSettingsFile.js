import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import DialogTitle from '@material-ui/core/DialogTitle'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogActions from '@material-ui/core/DialogActions'
import Dialog from '../Dialog'

const InvalidSettingsFile = ({ acknowledgeInvalidFile, invalidSettingsFile }) => (
  <Dialog
    onRequestClose={acknowledgeInvalidFile}
    open={!!invalidSettingsFile}
  >
    <DialogTitle>Error During Restore</DialogTitle>
    <DialogContent>
      <DialogContentText />
    </DialogContent>
    <DialogActions>
      <Button onClick={acknowledgeInvalidFile}>Close</Button>
    </DialogActions>
  </Dialog>
)

InvalidSettingsFile.propTypes = {
  acknowledgeInvalidFile: PropTypes.func,
  invalidSettingsFile: PropTypes.string,
}

InvalidSettingsFile.defaultProps = {
  acknowledgeInvalidFile: null,
  invalidSettingsFile: null,
}

export default InvalidSettingsFile
