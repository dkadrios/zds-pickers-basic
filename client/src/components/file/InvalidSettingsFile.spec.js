import InvalidSettingsFile from './InvalidSettingsFile'

const acknowledgeInvalidFile = jest.fn()

const props = {
  acknowledgeInvalidFile,
}

describe('InvalidSettingsFile tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(InvalidSettingsFile)(props).toMatchSnapshot()
  })
})
