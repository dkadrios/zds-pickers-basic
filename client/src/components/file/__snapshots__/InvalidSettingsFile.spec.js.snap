// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidSettingsFile tests renders correctly 1`] = `
<Dialog
  onRequestClose={[MockFunction]}
  open={false}
>
  <WithStyles(DialogTitle)>
    Error During Restore
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText) />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      Close
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</Dialog>
`;
